katsdpdata/met_handler.py

```diff
@@ -33,6 +33,111 @@ def __init__(self, solr_url, product_type, product_name, product_id=None):
         else:
             self.product_id = str(uuid.uuid4())
 
+    def get_prod_met(self, prod_id=None):
+        """Get the metadata for a product, if prod_id is not specified, use self.prod_id.
+
+        Parameters
+        ----------
+        prod_id: string : the product id to find.
+
+        Returns
+        -------
+        met: dict : metadata containing the _version_ for version tracking commits to solr.
+        """
+        copy_fields = ['Observer_lowercase']
+        if not prod_id:
+            prod_id = self.prod_id
+        query = 'id:{}'.format(prod_id)
+        res = self.solr.search(query)
+        if res.hits == 0:
+            return None
+        elif res.hits > 1:
+            raise MetaDataHandlerException('{} returned for {}'.format(res.hits, query))
+        # filter out copy field
+        doc = res.docs[0]
+        for c in copy_fields:
```

Comment:
We could remove '_version_' as well. The only thing I ever do with it is remove it before updating a product. Although i guess it might be worth checking the _version_ of a product before updating it to make sure you aren't clobbering another change. I don't do this anywhere though.

Reply:
During debugging, the version checking has been nice to spot errors, so I'll leave this in for now.